core/src/main/java/org/apache/brooklyn/entity/stock/AsyncApplicationImpl.java

```diff
@@ -0,0 +1,473 @@
+import org.apache.brooklyn.core.BrooklynLogging;
+import org.apache.brooklyn.core.config.BasicConfigInheritance;
+import org.apache.brooklyn.core.config.ConfigKeys;
+import org.apache.brooklyn.core.entity.AbstractApplication;
+import org.apache.brooklyn.core.entity.Attributes;
+import org.apache.brooklyn.core.entity.Entities;
+import org.apache.brooklyn.core.entity.lifecycle.Lifecycle;
+import org.apache.brooklyn.core.entity.lifecycle.ServiceStateLogic;
+import org.apache.brooklyn.core.entity.lifecycle.ServiceStateLogic.ServiceProblemsLogic;
+import org.apache.brooklyn.enricher.stock.AbstractMultipleSensorAggregator;
+import org.apache.brooklyn.util.collections.MutableList;
+import org.apache.brooklyn.util.collections.MutableSet;
+import org.apache.brooklyn.util.collections.QuorumCheck;
+import org.apache.brooklyn.util.exceptions.Exceptions;
+import org.apache.brooklyn.util.text.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.reflect.TypeToken;
+
+public class AsyncApplicationImpl extends AbstractApplication implements AsyncApplication {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AsyncApplicationImpl.class);
+
+    @Override
+    public void init() {
+        // Code below copied from BasicAppliationImpl.
+        // Set the default name *before* calling super.init(), and only do so if we don't have an 
+        // explicit default. This is a belt-and-braces fix: before we overwrote the defaultDisplayName
+        // that was inferred from the catalog item name.
+        if (Strings.isBlank(getConfig(DEFAULT_DISPLAY_NAME))) {
+            setDefaultDisplayName("Application ("+getId()+")");
+        }
+        super.init();
+    }
+    
+    @Override
+    protected void initEnrichers() {
+        // Deliberately not calling `super.initEnrichers()`. For our state (i.e. "service.state" 
+        // and "service.isUp"), we rely on the `serviceStateComputer`. This keeps things a lot
+        // simpler. However, it means that if someone manually sets a "service.notUp.indicators" 
+        // or "service.problems" then that won't cause the entity to transition to false or ON_FIRE.
+        
+        enrichers().add(EnricherSpec.create(ServiceStateComputer.class)
+                .configure(ServiceStateComputer.FROM_CHILDREN, true)
+                .configure(ServiceStateComputer.UP_QUORUM_CHECK, config().get(UP_QUORUM_CHECK))
+                .configure(ServiceStateComputer.RUNNING_QUORUM_CHECK, config().get(RUNNING_QUORUM_CHECK)));
+
+    }
+
+    // Only overrides AbstractApplication.start so as to disable the publishing of expected=running
+    @Override
+    public void start(Collection<? extends Location> locations) {
+        this.addLocations(locations);
+        // 2016-01: only pass locations passed to us, as per ML discussion
```

Comment:
Copy-pasted from AbstractApplication.start. I'll add to the method's comment:
`Code is copy-pasted verbatim from AbstractAppliation, except for not setting Lifecycle.RUNNING.`